File: 卡码/代码随想录/1/06-二叉树/C++/01-二叉树遍历方式/LeetCode_0117.cpp
//117. 填充每个节点的下一个右侧节点指针
#include <iostream>
#include <queue>
using namespace std;

class Node {
public:
  int val;
  Node* left;
  Node* right;
  Node* next;
  Node() : val(0), left(NULL), right(NULL), next(NULL) {}
  Node(int _val) : val(_val), left(NULL), right(NULL), next(NULL) {}
  Node(int _val, Node* _left, Node* _right, Node* _next):val(_val), left(_left), right(_right), next(_next) {}
};

class Solution {
public:
  Node* connect(Node* root) {
  	queue<Node*> q;
  	int size=0;
  	Node *cur,*prev;
  	if(root) q.push(root);
  	while(!q.empty()){
			size=q.size();
			for(int i=0;i<size;i++){
				if(i==0){
					prev=q.front();
					q.pop();
					cur=prev;
				}else{
					cur=q.front();
					prev->next=cur;
					prev=cur;
					q.pop();
				}
				if(cur->left) q.push(cur->left);
				if(cur->right) q.push(cur->right);
			}
			cur->next=nullptr;
		}
		return root;
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/二叉树/_01_二叉树遍历方式/LeetCode_0117.java
package com.kurumi.codeThoughts.二叉树._01_二叉树遍历方式;

import java.util.LinkedList;
import java.util.Queue;

import com.kurumi.codeThoughts.entity.Node;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 117. 填充每个节点的下一个右侧节点指针 II
 */
public class LeetCode_0117 {
    static class Solution {
        public Node connect(Node root) {
            Queue<Node> q = new LinkedList<>();
            Node pre = null, cur = null;
            if (root != null) {
                q.offer(root);
            }
            while (!q.isEmpty()) {
                int size = q.size();
                for (int i = 0; i < size; i++) {
                    cur = q.poll();
                    if (i != 0) {
                        pre.next = cur;
                    }
                    pre = cur;
                    if (cur.left != null) {
                        q.offer(cur.left);
                    }
                    if (cur.right != null) {
                        q.offer(cur.right);
                    }
                }
                cur.next = null;
            }
            return root;
        }
    }
}


